neuraxle/checkpoints.py

```diff
@@ -91,49 +99,76 @@ class PickleCheckpointStep(BaseCheckpointStep):
     to eventually be able to load them using the checkpoint pipeline runner.
     """
 
-    def __init__(self, force_checkpoint_name: str = None, cache_folder: str = DEFAULT_CACHE_FOLDER):
-        super().__init__(force_checkpoint_name)
+    def __init__(self, cache_folder: str = DEFAULT_CACHE_FOLDER):
+        super().__init__()
         self.cache_folder = cache_folder
-        self.force_checkpoint_name = force_checkpoint_name
 
-    def read_checkpoint(self, data_inputs):
+    def read_checkpoint(self, data_container: DataContainer):
         """
         Read pickle files for data inputs and expected outputs checkpoint
         :return: tuple(data_inputs, expected_outputs
         """
-        data_inputs_checkpoint_file_name = self.checkpoint_path
-        with open(self.get_checkpoint_file_path(data_inputs_checkpoint_file_name), 'rb') as file:
-            data_inputs = pickle.load(file)
-
-        return data_inputs
-
-    def save_checkpoint(self, data_inputs):
+        checkpoint_data_container = DataContainer(
+            current_ids=[],
+            data_inputs=[],
+            expected_outputs=None
+        )
+
+        for current_id, data_input, expected_output in zip(
```

Comment:
Please open an issue or create right now the Streaming classes of the checkpoints + Streaming wrapper as block checkpoint steps.